tools/mason/MasonSearch.chpl

```diff
@@ -0,0 +1,78 @@
+
+use MasonHelp;
+use MasonUpdate;
+
+use FileSystem;
+use Regexp;
+
+//
+// TODO:
+// - order results by some kind of 'best match' metric
+// - allow regex searches
+// - allow for exclusion of a pattern
+//
+
+proc masonSearch(origArgs : [] string) {
+  var args : [1..origArgs.size] string = origArgs;
+
+  const printHelp = || reduce [a in args] (a == "-h" || a == "--help");
+  if printHelp {
+    masonSearchHelp();
+    exit(0);
+  }
+
+  updateRegistry("", args);
+
+  consumeArgs(args);
+
+  // If no query is provided, list all packages in registry
+  const query = if args.size > 0 then args.tail().toLower()
+                else ".*";
+
+  const searchDir = MasonUpdate.getRegistryDir() + "/Bricks/";
+  const pattern   = compile(query, ignorecase=true);
+
+  var results : [1..0] string;
+  for dir in listdir(searchDir, files=false, dirs=true) {
```

Comment:
Is this searching all packages or only the subset of packages that supports the current compiler version? It's not clear to me which it should search, but it seems like something that should be toggle-able via flag. Not necessary for version `0.1.0`, but we *should* make it clear in the `--help` output.

Reply:
All packages for now.

Reply:
Mind making that more explicit in the help documentation?